byzcoin/bcadmin/main.go

```diff
@@ -1062,6 +1069,95 @@ func darcShow(c *cli.Context) error {
 	return err
 }
 
+func debugFollow(c *cli.Context) error {
+	if c.NArg() < 1 {
+		return errors.New("please give the following arguments: url [bcID]")
+	}
+	if c.NArg() == 1 {
+		return debugList(c)
+	}
+
+	r := &onet.Roster{List: []*network.ServerIdentity{{URL: c.Args().First()}}}
+	if r == nil {
+		return errors.New("couldn't create roster")
+	}
+	bcID, err := hex.DecodeString(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+
+	local := onet.NewLocalTest(cothority.Suite)
+	defer local.CloseAll()
+	servers := local.GenServers(1)
+	s := servers[0].Service(byzcoin.ServiceName).(*byzcoin.Service)
+
+	cl := skipchain.NewClient()
+	var sst *byzcoin.StagingStateTrie
+	sb, err := cl.GetSingleBlock(r, bcID)
+	if err != nil {
+		return err
+	}
+	for {
```

Comment:
Can you move all that logic to the byzcoin Client (api.go) so that we can keep the memory staging trie as private ?

Reply:
Hmm - difficult. If I move it all to `api.go`, I don't see a nice way to handle the `LocalTest` I'm using. Also, I think being able to replay things outside of ByzCoin makes also sense to other projects, e.g., if somebody wants to re-create the state of a given block. So I'd rather vote to have this as a useful export of the memory staging trie.

Reply:
What's wrong with using `LocalTest` there ?
Well it's funny because I had the same thinking so this piece of code could be available with the Client (instead of being private in bcadmin) so other projects don't have to deal with the logic of the trie.

Reply:
After talking with Jeff, we agreed that the logic inside the for loop should be a function provided by the byzcoin service. The function would take a callback that returns a block like the line 1096. Does it make sense ?

Reply:
No, I don't see how and why you want to add a callback in the byzcoin service. Also, should this be authenticated like I proposed?

Reply:
Closing this conversation - please have a look at #1858